### ISSUE TITLE
Drush Sanitize fails on [error]  No entity bundle was specified

### ISSUE THREAD

**Issue by brooke-heaton** [drush-ops/drush#3636]
**Describe the bug**
When running `drush sanitize`, it works through a series of fields but always fails at `[error]  No entity bundle was specified` - it is unclear on why this happens. 

**To Reproduce**
`drush sanitize` with no whitelist

**Expected behavior**
I expect a clean sanitization without error.

**Actual behavior**
It fails and triggers an error that seems entirely unrelated to users or text fields.

**Workaround**
No.

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 9.4-dev-dev
| Drupal version? | 8.5
| PHP version     | 7.1
| OS?             | Mac

**Additional information**
I am using Acquia BLT so the sanitization is part of BLT.

**Comment by brooke-heaton** [comment]
Fair enough. I probably won't have the time. I stepped through it a few weeks ago and did find that it was likely related to a commerce product variation type - that was the entity without a bundle. But how that would ever appear due to a user text file sanitization is beyond me.

**Comment by ey-** [comment]
https://github.com/drush-ops/drush/blob/master/src/Drupal/Commands/sql/SanitizeUserFieldsCommands.php#L69

It looks like this is too early, as this generates first sample values for _all_ user fields, also entity references for example. And later on the generated values are only used by the listed field types in the switch.
